Makefile

```diff
@@ -77,166 +50,52 @@ endif
-	 $(COMPARE_OBJ) && \
-	 touch -r $(BASE_DIR)/$(@F:.marker=.o) $(@:.marker=.o), \
-	 echo "Object file $(@F:.marker=.o) is not in the current archive" \
-	)
-endif
-ifneq ($(MODERN_LD),0)
-	tools/set_o32abi_bit.py $(WORKING_DIR)/$(@:.marker=.o)
-	$(CROSS)strip $(WORKING_DIR)/$(@:.marker=.o) -N asdasdasdasd
-	$(CROSS)objcopy --remove-section .mdebug $(WORKING_DIR)/$(@:.marker=.o)
-endif
-# create or update the marker file
-	@touch $@
-
-$(S_MARKER_FILES): $(BUILD_DIR)/%.marker: %.s
-	cd $(<D) && $(AS) $(ASFLAGS) $(MIPS_VERSION) $(CPPFLAGS) $(ASOPTFLAGS) $(<F) $(IINC) -o $(WORKING_DIR)/$(@:.marker=.o)
-ifneq ($(COMPARE),0)
-# check if this file is in the archive; patch corrupted bytes and change file timestamps to match original if so
-	@$(if $(findstring $(BASE_DIR)/$(@F:.marker=.o), $(BASE_OBJS)), \
-	 python3 tools/fix_objfile.py $(@:.marker=.o) $(BASE_DIR)/$(@F:.marker=.o) $(STRIP) && \
-	 $(COMPARE_OBJ) && \
-	 touch -r $(BASE_DIR)/$(@F:.marker=.o) $(@:.marker=.o), \
-	 echo "Object file $(@F:.marker=.o) is not in the current archive" \
-	)
-endif
-ifneq ($(MODERN_LD),0)
-	tools/set_o32abi_bit.py $(WORKING_DIR)/$(@:.marker=.o)
-	$(CROSS)strip $(WORKING_DIR)/$(@:.marker=.o) -N asdasdasdasd
-	$(CROSS)objcopy --remove-section .mdebug $(WORKING_DIR)/$(@:.marker=.o)
-endif
-# create or update the marker file
-	@touch $@
-
-# Rule for building files that require specific file paths in the mdebug section
-$(MDEBUG_FILES): $(BUILD_DIR)/src/%.marker: src/%.s
-	cp $(<:.marker=.s) $(dir $@)
-	mkdir -p $(@:.marker=)
-	export USR_INCLUDE=$(WORKING_DIR)/include && cd $(@:.marker=) && $(AS) $(ASFLAGS) $(CPPFLAGS) ../$(<F) -I/usr/include -o $(notdir $(<:.s=.o))
-	mv $(@:.marker=)/$(<F:.s=.o) $(@:.marker=)/..
-ifneq ($(COMPARE),0)
-# check if this file is in the archive; patch corrupted bytes and change file timestamps to match original if so
-	@$(if $(findstring $(BASE_DIR)/$(@F:.marker=.o), $(BASE_OBJS)), \
-	 python3 tools/fix_objfile.py $(@:.marker=.o) $(BASE_DIR)/$(@F:.marker=.o) && \
-	 $(COMPARE_OBJ) && \
-	 touch -r $(BASE_DIR)/$(@F:.marker=.o) $(@:.marker=.o), \
-	 echo "Object file $(@F:.marker=.o) is not in the current archive" \
-	)
-endif
-ifneq ($(MODERN_LD),0)
-	tools/set_o32abi_bit.py $(WORKING_DIR)/$(@:.marker=.o)
-	$(CROSS)strip $(WORKING_DIR)/$(@:.marker=.o) -N asdasdasdasd
-	$(CROSS)objcopy --remove-section .mdebug $(WORKING_DIR)/$(@:.marker=.o)
-endif
-# create or update the marker file
-	@touch $@
+$(BUILD_DIR)/src/gu/parse_gbi.o: GBIDEFINE := -DF3D_GBI
+$(BUILD_DIR)/src/gu/us2dex_emu.o: GBIDEFINE :=
+$(BUILD_DIR)/src/gu/us2dex2_emu.o: GBIDEFINE :=
+$(BUILD_DIR)/src/sp/sprite.o: GBIDEFINE := -DF3D_GBI
+$(BUILD_DIR)/src/sp/spriteex.o: GBIDEFINE :=
+$(BUILD_DIR)/src/sp/spriteex2.o: GBIDEFINE :=
+$(BUILD_DIR)/src/voice/%.o: OPTFLAGS += -DLANG_JAPANESE -I$(WORKING_DIR)/src -I$(WORKING_DIR)/src/voice
+$(BUILD_DIR)/src/voice/%.o: CC := $(WORKING_DIR)/tools/compile_sjis.py -D__CC=$(CC) -D__BUILD_DIR=$(BUILD_DIR)
```

Comment:
On modern gcc this may be able to use `-fexec-charset=sjis` but it would make sense if you want to leave this to its own contribution